### PR TITLE
 Migrate sm_kinematics to pure CMake

### PR DESCRIPTION
**Pull request by frneer**
Migrate the package to pure CMake and add it to the list of packages tested in the CI workflow.

### CODE REVIEW

File: Schweizer-Messer/sm_kinematics/test/QuaternionTests.cpp [hunk 1]
```diff
@@ -104,6 +104,10 @@ TEST(QuaternionAlgebraTestSuite, testAxisAngle2QuatAndBack)
       sm::eigen::assertNear(q, axisAngle2quat(quat2AxisAngle(q)), q.block<3,1>(0, 0).norm() * eps * 20, SM_SOURCE_FILE_POS, "");
 
       // Create a random point in R3
+      // Set a known seed to get reproducible results, since some edge cases can turn the quaternion (-q), which
+      // represents the same rotation, but has a different axis-angle representation.
+      // To improve this test we need to extract the "near" concept and assert that is near v_b or -v_b.
+      std::srand((unsigned int) time(0));
```
[the remark this answers is not in the document]
Comment (frneer) [reply Schweizer-Messer/sm_kinematics/test/QuaternionTests.cpp]: Added a patch for this. @hidmic

**Comment by frneer** [comment]
@hidmic rebased to fix merge conflicts, I think this one is ready to go PTAL

**Review by hidmic — APPROVED** [review]
